homeassistant/components/technove/binary_sensor.py

```diff
@@ -0,0 +1,102 @@
+"""Support for TechnoVE binary sensor."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from technove import Station as TechnoVEStation
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TechnoVEDataUpdateCoordinator
+from .entity import TechnoVEEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class TechnoVEBinarySensorDescription(BinarySensorEntityDescription):
+    """Describes TechnoVE binary sensor entity."""
+
+    value_fn: Callable[[TechnoVEStation], bool | None]
+
+
+BINARY_SENSORS = [
+    TechnoVEBinarySensorDescription(
+        key="conflict_in_sharing_config",
+        translation_key="conflict_in_sharing_config",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.conflict_in_sharing_config,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="in_sharing_mode",
+        translation_key="in_sharing_mode",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.in_sharing_mode,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="is_battery_protected",
+        translation_key="is_battery_protected",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.is_battery_protected,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="is_session_active",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=BinarySensorDeviceClass.BATTERY_CHARGING,
+        value_fn=lambda station: station.info.is_session_active,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="is_static_ip",
+        translation_key="is_static_ip",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.is_static_ip,
+    ),
+    TechnoVEBinarySensorDescription(
+        key="update_available",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=BinarySensorDeviceClass.UPDATE,
+        value_fn=lambda station: not station.info.is_up_to_date,
+    ),
```

Comment:
Will you also be able to update in the future?

Reply:
Maybe, but I don't think so. Not for now at least.

The mobile app offers OTA updates but the server to check for updates is not public, I would need to reverse engineer the mechanism.

Is there a better way than binary sensor you would recommend?

Reply:
In that case this will do just fine :)